src/Core/Jobs/JobFactory.cs

```diff
@@ -16,7 +18,7 @@ public JobFactory(IServiceProvider container)
     public IJob NewJob(TriggerFiredBundle bundle, IScheduler scheduler)
     {
         var scope = _container.CreateScope();
-        return scope.ServiceProvider.GetService(bundle.JobDetail.JobType) as IJob;
+        return (scope.ServiceProvider.GetService(bundle.JobDetail.JobType) as IJob)!;
```

Comment:
As I understand it this should only be null if a job type hasn't been registered in DI before being executed by Quartz.NET, which shouldn't happen and that'll just lead to Quartz throwing a bit after calling this anyway.

I haven't used Quartz before, but from looking into it a bit I think as of ~4 years ago this isn't necessary, as Quartz can handle all of this itself. It looks like there could be a pretty big cleanup of most of the use of Quartz, which would get rid of this, but that's a whole other thing.